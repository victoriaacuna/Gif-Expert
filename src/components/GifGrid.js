import React, { useState, useEffect } from 'react'
import { useFetchGif } from '../hooks/useFetchGif'
import { GifGridItem } from './GifGridItem';
// import { getGif } from '../helpers/getGif';

const GifGrid = ({category}) => {

    const {data:images, loading} = useFetchGif(category);

    // const [images, setImages] = useState([])

    // // Para que la funciÃ³n solo se ejecute una sola vez al renderizar el componente.
    // // Si le pusieras [category] en vez de [] lo que hace es que, cuando category cambie, es que se vuelve a ejecutar ese codigo.
    // useEffect(() => {
    //     getGif(category)
    //         .then(imgs => {
    //             setImages(imgs)
    //         });
    // }, []);


    


    return (
        <>
            <h3>{category}</h3>
            {loading?'Cargando...':'Ready to go'}
            <div className="card-grid">
            {
                images.map( img => (
                    <GifGridItem key={img.id} {...img}/>
                ))
            }
            </div>
        </>
    )
}



export default GifGrid
