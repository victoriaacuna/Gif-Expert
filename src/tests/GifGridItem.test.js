import '@testing-library/jest-dom';
import {shallow} from 'enzyme';
import {GifGridItem} from './../components/GifGridItem';
import React from 'react';

describe('Pruebas componente Gif Grid Item', () => {

    const title = 'un titulo';
    const url = 'https://localhost:3000/algo.jpg';

    const wrapper = shallow(<GifGridItem url={url} title={title}/>);

    test('Should match snapshot', () => {

        expect(wrapper).toMatchSnapshot();

    });

    test('img should have url and alt', () => {
        const img = wrapper.find('img');
        console.log(img.props());
        console.log(img.prop('src'));
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    });

    test('should have className animate__bounce"', () => {
        const div = wrapper.find('div');
        const className = 'animate__bounce';
        console.log(div.hasClass(className));
        expect(div.hasClass(className)).toBe(true);

        // const className = div.prop('className');
        // expect(className.includes('animate__bounce')).toBe(true);
    })
    
    



});