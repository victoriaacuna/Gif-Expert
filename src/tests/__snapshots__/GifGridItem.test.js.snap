// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas componente Gif Grid Item Should match snapshot 1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img
    alt="un titulo"
    src="https://localhost:3000/algo.jpg"
  />
  <p>
    un titulo
  </p>
</div>
`;
